handler/gcp_test.go

```diff
@@ -4,25 +4,33 @@ import (
 	"context"
 	"net/http"
 	"net/http/httptest"
+	"os"
+	"strings"
 	"testing"
 
-	"github.com/go-resty/resty/v2"
+	"cloud.google.com/go/compute/metadata"
 	"github.com/stretchr/testify/require"
 )
 
 func TestGCPInterruptChecker(t *testing.T) {
-	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		require.Equal(t, "/computeMetadata/v1/instance/preempted", r.URL.String())
-
-		w.WriteHeader(http.StatusOK)
-		_, err := w.Write([]byte("TRUE"))
+	mux := http.NewServeMux()
```

Comment:
nitpick: would have provided mocked metadata client with a mocked `metadata.Get` method instead; by building HTTP server we're testing google SDK which is not desirable - tests break if client implementation changes, e.g. requires additional http header.